samples/routing/custom/sample.md

```diff
@@ -0,0 +1,112 @@
+ 1. Hit `<enter>` several times to send more messages.
+ 1. Notice that only every second message gets processed by Sales.1. Client still does not know that Sales.2 is down.
+ 1. Wait until consoles show that Sales.2 heartbeat timed out.
+ 1. Hit `<enter>` several times to send more messages. 
+ 1. Notice that all orders are now routed to Sales.1 queue.
+
+
+### Recovery
+
+ 1. In Visual Studio right-click on `Sales2` project and select `Debug -> Start new instance`.
+ 1. Notice that all messages sent to Sales.2 while it was down are now processed.
+ 1. Wait until other endpoints detect Sales.2 again.
+ 1. Hit `<enter>` several times to send more messages.
+ 1. Notice that orders are again routed to both Sales instances in round-robin fashion.
+
+
+## Code walk-through
+
+This sample contains four projects:
+
+ * Shared - A class library containing common routing code including the message definitions.
+ * Client - A console application responsible for sending the initial `PlaceOrder` message.
+ * Sales - A console application responsible for processing the `PlaceOrder` command and generating `OrderAccepted` event.
+ * Shipping - A console application responsible for processing the `OrderAccepted` message.
+ * Billing - Another console application responsible for processing the `OrderAccepted` message.
+
+
+### Client
+
+The Client does not store any data. It mimics the front-end system where orders are submitted by the users and passed via the bus to the back-end.
+
+
+### Sales
+
+The Sales project mimics the back-end system where orders are accepted. Apart from the standard NServiceBus configuration it enables the custom automatic routing:
+
+snippet:EnableAutomaticRouting
+
+NOTE: In order to use this custom routing all published types need to be specified.
+
+In real-world scenarios NServiceBus is scaled out by deploying multiple physical instances of a single logical endpoint to multiple machines (e.g. `Sales` in this sample). For simplicity in this sample the scale-out is simulated by having two separate endpoints `Sales` and `Sales2`.
+
+
+### Shipping and Billing
+
+Shipping and Billing mimic other back-end systems.
+
+
+### Shared project
+
+The shared project contains definitions for messages and the custom routing logic. 
+
+
+### Custom automatic routing
+
+The automatic routing is based on the idea of endpoints exchanging information about types of messages they handle and types of messages they publish. They do it via a table in SQL Server database (each endpoint instance own one row keyed by this endpoint's transport address). Following code is used inside the custom automatic routing feature to wire up all its components:
+
+snippet:Feature
+
+It creates a publisher and a subscriber for the routing information along with the communication object they use to exchange the information between endpoint instances. It also registers an additional behavior that makes sure all published types are properly advertised.
```

Comment:
> creates a publisher and a subscriber for the routing information

Why do I need those? Does one instance of each live in every endpoint? How does it work?

Would diagram help?